### PR TITLE
chore: fixes on the dark trails quest 

### PR DESCRIPTION
**Pull request by dudantas**
fixes on: dark trails quest
refactoring some scripts
move to new TFS API 1.0
switch to revscriptsys
* note that some scripts have not yet been refactored, just moved to revscriptsys

Some npcs are without scripts:
* Barnabas Dee
* Quandon Ghost
* Terrence

if anyone has the scripts please do PR

It's fixes the #547 

thanks for @Raphaellb for testing and correcting the scripts, if you can test again, I will be thankful

### CODE REVIEW

File: data/npc/scripts/Barazbaz.lua [hunk 1]
```diff
@@ -34,13 +34,14 @@ function creatureSayCallback(cid, type, msg)
...
         player:setStorageValue(20059, 0)
  
-    elseif(msgcontains(msg, "notebook")) and player:getStorageValue(10050) == 10 then
-        npcHandler:say(" I know that handwriting you describe! It belongs to a traveller from far away. Magistrate Sholley introduced him to me and she was quite excited to learn more about our city's past. ...", cid)
-        npcHandler:say("I should have thought of him right in the beginning when I heard the stuff you mentioned. But I haven't seen him for a while. You should ask Sholley about her friend to learn about his whereabouts.", cid)
-        player:setStorageValue(10050, 12)
-        player:setStorageValue(30050, 1)
-        player:setStorageValue(30051, 0)
-    else
+	elseif(msgcontains(msg, "notebook")) and (player:getStorageValue(10050) == 10 and getPlayerItemCount(cid, 12406) == 1) then
```
Comment (gpedro) [inline data/npc/scripts/Barazbaz.lua]: can you export thisstorages numbers to `051-storages`?

File: data/npc/scripts/Barazbaz.lua [hunk 2]
```diff
@@ -34,13 +34,14 @@ function creatureSayCallback(cid, type, msg)
...
+		npcHandler:say(" I know that handwriting you describe! It belongs to a traveller from far away. Magistrate Sholley introduced him to me and she was quite excited to learn more about our city's past. ...", cid)
+		npcHandler:say("I should have thought of him right in the beginning when I heard the stuff you mentioned. But I haven't seen him for a while. You should ask Sholley about her friend to learn about his whereabouts.", cid)
+		player:setStorageValue(10050, 12)
+		player:setStorageValue(30050, 1)
+		player:setStorageValue(30051, 0)
```
Comment (gpedro) [inline data/npc/scripts/Barazbaz.lua]: can you export thisstorages numbers to `051-storages`?

Comment (dudantas) [reply data/npc/scripts/Barazbaz.lua]: yes sir

File: data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+function death_priest.onKill(player, target, lastHit)
+config = {
+monster = "Death Priest Shargon",
+storage = Storage.DarkTrails.Mission18,
+}
+
+	local creature = Creature(cid)
+	if isMonster(target) then
+		if creature:getName(target) == config.monster and player:getStorageValue(config.storage) < 1 then
```
Comment (vankk) [inline data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua]: `Creature:getName(target)` won't work. Try replace that with `target:getName()`

File: data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua [hunk 4]
```diff
@@ -0,0 +1,18 @@
+local death_priest = CreatureEvent("Death Priest onKill")
+
+function death_priest.onKill(player, target, lastHit)
+config = {
```
Comment (vankk) [inline data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua]: This table with storage and name is really needed?

File: data/scripts/creatureevents/quests/dark trails/the_ravager_storage.lua [hunk 5]
```diff
@@ -0,0 +1,18 @@
+local the_ravager = CreatureEvent("The Ravager onKill")
+
+function the_ravager.onKill(player, target, lastHit)
+config = {
+monster = "The Ravager",
```
Comment (vankk) [inline data/scripts/creatureevents/quests/dark trails/the_ravager_storage.lua]: Same here.

File: data/scripts/creatureevents/quests/dark trails/the_ravager_storage.lua [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+storage = Storage.DarkTrails.Mission11,
+}
+
+	local creature = Creature(cid)
+	if isMonster(target) then
+		if creature:getName(target) == config.monster and player:getStorageValue(config.storage) < 1 then
```
Comment (vankk) [inline data/scripts/creatureevents/quests/dark trails/the_ravager_storage.lua]: Same here.

File: data/npc/scripts/Quandons Ghost.lua [hunk 7]
```diff
@@ -2,9 +2,20 @@ local keywordHandler = KeywordHandler:new()
...
-function onThink()				npcHandler:onThink()					end
+function onCreatureAppear(cid)			npcHandler:onCreatureAppear(cid) 			end
+function onCreatureDisappear(cid)		npcHandler:onCreatureDisappear(cid) 		end
+function onCreatureSay(cid, type, msg)	npcHandler:onCreatureSay(cid, type, msg) 	end
+function onThink()						npcHandler:onThink() 						end
+function onThink() 						npcHandler:onThink() 						end
 
 npcHandler:addModule(FocusModule:new())
+
+function creatureSayCallback(cid, type, msg)
```
Comment (vankk) [inline data/npc/scripts/Quandons Ghost.lua:13]: Is neeed that function that isn't been used for now?

File: data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua (hunk 4 above)
Comment (dudantas) [reply data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua]: What?
This is the variable for revscriptsys register action without action.xml

Comment (dudantas) [reply data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua]: No, i'm go remove this table. Thanks.

File: data/npc/scripts/Barazbaz.lua (hunk 1 above)
Comment (dudantas) [reply data/npc/scripts/Barazbaz.lua]: fixed

File: data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua (hunk 3 above)
Comment (dudantas) [reply data/scripts/creatureevents/quests/dark trails/death_priest_storage.lua]: i go try, thanks for the suggestion =D

File: data/npc/scripts/Quandons Ghost.lua (hunk 7 above)
Comment (dudantas) [reply data/npc/scripts/Quandons Ghost.lua:13]: As the code is under development, let to me create npc in the next few days.

**Pull request closed without merging**